### DISCUSSION TITLE
Preserving Temporary Files from Simulator after XCTest with Fastlane Scan (removing XCTestDevices folder)

### DISCUSSION THREAD

**Discussion by sou1t** [fastlane/fastlane#21655] [Q&A]
Hello Fastlane Community,

I'm currently facing a challenge with XCTest in my CI/CD pipeline using Fastlane's scan tool. After running my tests, I need to access and preserve specific temporary files created in the simulator's directory (typically located at ~/Library/Developer/XCTestDevices/{device_id}/data/Containers/Data/Application/{app_id}/tmp). However, these files seem to be deleted automatically after the tests complete when run via Fastlane.

Here's my situation:

- I'm using Fastlane scan to run my tests.
- Post-test, the temporary simulator directory is cleared, removing the files I need.
- I need to copy these files to a project directory for further use.

My `run_tests` looks like

``` ruby
    run_tests(
      buildlog_path: './reports/tests/',
      code_coverage: true,
      concurrent_workers: 10,
      derived_data_path: DERIVED_DATA_PATH,
      destination: "name=#{Constants::DEVICE},OS=#{Constants::SIMULATOR_VERSION}",
      max_concurrent_simulators: 10,
      number_of_retries: 3,
      result_bundle: true,
      output_directory: "./build/.reports/#{options[:scheme]}",
      output_types: 'junit',
      scheme: options[:scheme],
      test_without_building: true,
      xcargs: {
        '-parallel-testing-enabled' => 'YES'
      }
    )
```

My questions are:

- Is there a way to configure Fastlane scan to prevent the deletion of these temporary files after test completion?
- Where I should insert a custom script or step within my Fastlane setup to copy these files before they are deleted?
- If anyone has faced a similar issue, how did you resolve it?
- Any guidance or suggestions on this matter would be greatly appreciated. I'm looking for the best practices to handle this situation effectively without disrupting the standard testing workflow.

Additionally, I would like to mention that when tests are run directly from Xcode, the temporary simulator directory remains intact between test runs, and no files are deleted. This behavior is different from what I experience when running tests via Fastlane scan. Understanding why there is a difference in file retention between running tests in Xcode and via Fastlane would also be helpful in addressing my issue.

Thank you in advance for your help and insights!

**Comment by formacion-tyris** [comment]
I'm in the same exact situation. Any tips?
